spec/validation/configure_spec.rb

```diff
@@ -247,6 +247,17 @@ def run_configure_validation(my_hash = {})
                                   })
       expect(run_configure_validation(h).keys).to eq([:default_integer_type])
     end
+
+    it 'fails if the default is an empty string' do
```

Comment:
I don't follow why we need a special case for empty string defaults - can't they be covered by the same validating as non-empty strings? Then as I understand it we wouldn't need this test or special handling for empty strings, as this is just another test for `default_type?`. Am I missing something? This validations stuff still looks like magic to me.

Reply:
Because of the crazy logic that occurs in the `Validation` I can't guarantee that empty strings will be treated the same way as a regular string. The new `Validator` class in a latter PR handles it better.